rhino/src/main/java/org/mozilla/javascript/ScriptableObject.java

```diff
@@ -2695,6 +2781,17 @@ private static LambdaSlot ensureLambdaSlot(Object name, int index, Slot existing
         }
     }
 
+    private static LambdaAccessorSlot ensureLambdaAccessorSlot(
```

Comment:
Looks like we're not using this method any more... FWIW, I was using this pattern as a simple way to switch slot classes in a clean way, and then doing the calculation after that, so the way I did this in other methods is a bit different from what you did, but it's all good.

But this is unused code so we should delete it!

Reply:
Thanks, I missed that one. It's gone.

Reply:
> I was using this pattern as a simple way to switch slot classes in a clean way, and then doing the calculation after that, so the way I did this in other methods is a bit different from what you did, but it's all good.

The version of`ensureLambdaAccessorSlot` I ended up using is closer to this [lambda function](https://github.com/mozilla/rhino/blob/7369f3fcd16d7925b9ca0d43e96100b1708cbdff/rhino/src/main/java/org/mozilla/javascript/ScriptableObject.java#L1614) in `defineOwnProperty`. It's not just (re)creating slot classes, but also doing validation, etc. If you think it's doing to much in a single function or could be named better, I'm happy to refactor this too @gbrail . 


Although maybe in next PR? :)